tephra-core/src/test/java/co/cask/tephra/persist/HDFSTransactionLogTest.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright © 2012-2014 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.tephra.persist;
+
+import co.cask.tephra.TxConstants;
+import co.cask.tephra.metrics.MetricsCollector;
+import co.cask.tephra.metrics.TxMetricsCollector;
+import co.cask.tephra.snapshot.SnapshotCodecV2;
+import co.cask.tephra.util.TransactionEditUtil;
+import com.google.common.io.Closeables;
+import com.google.common.primitives.Longs;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.MiniDFSCluster;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.SequenceFile;
+import org.apache.hadoop.io.Text;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+
+
+/**
+ * Testing for complete and partial sycs of {@link TransactionEdit} to {@link HDFSTransactionLog}
+ */
+public class HDFSTransactionLogTest {
+  @ClassRule
+  public static TemporaryFolder tmpFolder = new TemporaryFolder();
+  private static final String LOG_FILE_PREFIX = "txlog.";
+
+  private static MiniDFSCluster dfsCluster;
+  private static Configuration conf;
+  private static MetricsCollector metricsCollector;
+
+  @BeforeClass
+  public static void setupBeforeClass() throws Exception {
+    Configuration hConf = new Configuration();
+    hConf.set(MiniDFSCluster.HDFS_MINIDFS_BASEDIR, tmpFolder.newFolder().getAbsolutePath());
+
+    dfsCluster = new MiniDFSCluster.Builder(hConf).numDataNodes(1).build();
+    conf = new Configuration(dfsCluster.getFileSystem().getConf());
+    metricsCollector = new TxMetricsCollector();
+  }
+
+  @AfterClass
+  public static void tearDownAfterClass() throws Exception {
+    dfsCluster.shutdown();
+  }
+
+  protected Configuration getConfiguration() throws IOException {
+    // tests should use the current user for HDFS
+    conf.unset(TxConstants.Manager.CFG_TX_HDFS_USER);
+    conf.set(TxConstants.Manager.CFG_TX_SNAPSHOT_DIR, tmpFolder.newFolder().getAbsolutePath());
+    conf.set(TxConstants.Persist.CFG_TX_SNAPHOT_CODEC_CLASSES, SnapshotCodecV2.class.getName());
```

Comment:
I see that all `TransactionStateStorage` tests use v2 codec. Can you please change all of them to use the latest codec?